matching-core/src/test/java/com/graphhopper/matching/MapMatchingTest.java

```diff
@@ -80,17 +79,23 @@ public MapMatchingTest(String name, TestGraphHopper hopper, AlgorithmOptions alg
         hopper.setEncodingManager(new EncodingManager(encoder));
         hopper.importOrLoad();
 
+        // Reduce penalty to allow U-turns in testUTurns
+        final double HEADING_PENALTY = 50;
+
         // force CH
         AlgorithmOptions chOpts = AlgorithmOptions.start()
-                .maxVisitedNodes(40)
-                .hints(new PMap().put(Parameters.CH.DISABLE, false))
+                .maxVisitedNodes(1000)
+                .hints(new PMap().put(Parameters.CH.DISABLE, false)
+                        //TODO Fix that CH routing still uses default penalty of 300
```

Comment:
testUTurns for CH still fails because CH routing does not consider the set parameter value for Parameters.Routing.HEADING_PENALTY. @karussell, can you please help?